### DISCUSSION TITLE
Add HTTP Request extension

### DISCUSSION THREAD

**Discussion by federico-terzi** [espanso/espanso#1204] [Feature requests and Ideas]
I've noticed a lot of packages rely on `curl` to execute HTTP requests and obtain the expansion text.
It might be a good idea to introduce a "request" extension so that use-case is solved without the use of an external command

**Comment by ntotao** [comment]
I think this Feature could be interesting to let wttr package work again?
https://github.com/bradyjoslin/espanso-package-wttr/issues/2

**Comment by ifallisequal** [comment]
For those still having issues with expansions that relied on the curl command when using Windows. There is a fairly easy workaround to restore this functionality. 

The issue appears to, at least in my case be, that after Microsoft disabled Internet Explorer it crippled curl (alias for Invoke-WebRequest) in older versions of Powershell in the process. A solution for this I have been using, is to install the latest version of Powershell "Powershell 7" then calling upon this new version by adding `pwsh -command` before curl e.g. `cmd: "pwsh -command curl '<URL>'"`.

**Comment by AucaCoyan** [comment]
Closing this discussion as #1935 is created to track the issue
